admin_organization_integration_test.go

```diff
@@ -13,6 +13,7 @@ import (
 )
 
 func TestAdminOrganizations_List(t *testing.T) {
+	checkTestNodeEnv(t)
```

Comment:
After reading the PR description about adding a `checkTestNodeEnv(t)` to all files, and seeing the impact, I'm wondering if it would be beneficial to explore another option for checking the test node environment. In the Ruby world, I would add a test class or module that does this checking, and all of the test classes would inherit from this new test class. I haven't done something similar in Go, so curious what your thoughts are about it? 🤔

Reply:
Absolutely! While we can't exactly have top-level tests inherit some general behavior, we can potentially introduce our own "test suite" to go-tfe, which would reduce boilerplate setup across tests. This would require a pretty sizable refactor of our tests currently. 

An example might look like:
```go
func TestWorkspacesList(t *testing.T) {
   // setupCloudTest() 
   // calls skipIfTFE(), checkTestNodeEnv(), etc
   // creates a go-tfe client
   // calls createOrganization, t.Cleanup(orgCleanup), upgrades an organization
  client, org := setupCloudTest(t) 

  // or alternatively

   // setupTFETest() 
   // calls skipIfCloud(), checkTestNodeEnv(), etc
   // creates a go-tfe client
   // calls createOrganization(), t.Cleanup(orgCleanup)
  client, org := setupTFETest(t) 
  
  // run rest of test logic here
  
}
```

Reply:
Could you add a follow on issue for us to refactor the go-tfe tests like you've described? I'm good with this PR addressing the test execution time, and then a subsequent PR addressing test refactoring.